Zimbra/zm-mailbox
# snickers: NICPS-32: add function to set appointment color

Fix : 
[UI]
- Added "color" button to Ajax/HTML/Mobile client
- Show appointment with the specified color

[Server]
- SOAP API change: color and rgb options are added to "comp" (components of invite)
- Added color data to Invite
- Added function to export/import calendar with COLOR property (RFC 7986)

Testing done : testing done by developer
- Appointment is shown with the color in each view
  - Day/Week/WorkWeek/Month view
  - print calendar/appointment
  - REST page. Note that if color param exists in the url, appointments are shown with the specified color.
- Create/Edit an appointment with color
- Create/Edit a recurring schedule and an exception with color
- If color is NONE, the appointment is shown with the folder color.
- Send an invitation to attendees WITHOUT color data
- Export ics with COLOR property
- Import ics: appointment is imported with with color
- Export/Import tgz addresses color data
- backup/restore: appointment is restored with color data at zmplayredo


Testing to be done by QA : testing needed to be done by QA in PS team

Note :
- appointment with custom color cannot be created on HTML/Mobil client
- custom color is not shown on Mobile client (appointment is shown with the folder color)
- custom color is not shown on REST page (appointment is shown with the folder color)
- custom color cannot be exported/imported. Supported colors: blue, cyan, green, purple, red, yellow, pink, gray, orange

Linked PR : 
https://github.com/Zimbra/zm-taglib/pull/5
https://github.com/Zimbra/zm-web-client/pull/250
